tensorflow/python/keras/callbacks.py

```diff
@@ -2745,14 +2745,13 @@ def __init__(self, filename, separator=',', append=False):
   def on_train_begin(self, logs=None):
     if self.append:
       if file_io.file_exists_v2(self.filename):
-        with open(self.filename, 'r' + self.file_flags) as f:
+        with file_io.FileIO(self.filename, 'r' + self.file_flags) as f:
           self.append_header = not bool(len(f.readline()))
       mode = 'a'
     else:
       mode = 'w'
-    self.csv_file = io.open(self.filename,
-                            mode + self.file_flags,
-                            **self._open_args)
```

Comment:
Sorry, somehow the comment wasn't preserved.

The self._open_args is removed in the new code, is that expected?

Reply:
yeah good point. It would be better if it stayed there, but `file_io.FileIO` only takes as input the `path` and the `mode`

Reply:
It seems that the flag was originally added in https://github.com/tensorflow/tensorflow/commit/c98ffffcb4e0cc668c0ff7b73d51677a7eb7dcf4#diff-acdb23c4fe8de16beef9cf6789ccd7da614f7f62e9b057cf841b19eadb624359R1362. 

Looking at https://docs.python.org/3/library/functions.html#open, I don't see a specific reason to have this flag changed (when writing/append to file). Let's submit this as is for now.